### PR TITLE
feat(developer): range warning for non-NFD chars 🙀 

### PR DESCRIPTION
**Pull request by srl295**
Fixes: #10316 

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/types/src/util/util.ts [hunk 1]
```diff
@@ -350,5 +353,36 @@ export class BadStringAnalyzer {
...
+/** Analyzer that checks if something isn't NFD */
+export class NFDAnalyzer extends StringAnalyzer {
+  protected analyzeCodePoint(c: string, ch: number): BadStringType {
+    const nfd = c.normalize("NFD");
+    if (c !== nfd) {
+      return BadStringType.denormalized;
+    } else {
+      return null;
+    }
+  }
```
Comment (mcdurdin) [inline common/web/types/src/util/util.ts:387]: I wonder about the performance of this -- seems like it could be quite expensive?

**Review by mcdurdin — APPROVED** [review]

**Comment by mcdurdin** [comment]
Our messages (errors, hints, and warnings) don't have line numbers at present, which is going to be quite painful for users!

File: common/web/types/src/util/util.ts (hunk 1 above)
Comment (srl295) [reply common/web/types/src/util/util.ts:387]: right. maybe some kind of cache would help.

Comment (srl295) [reply common/web/types/src/util/util.ts:387]: yes, could perhaps cache?

Comment (srl295) [reply common/web/types/src/util/util.ts:387]: it's only on actual ranges in transforms.

**Comment by srl295** [comment]
> Our messages (errors, hints, and warnings) don't have line numbers at present, which is going to be quite painful for users!

~what about...~ << moving it to a new ticket! #10636

File: common/web/types/src/util/util.ts (hunk 1 above)
Comment (mcdurdin) [reply common/web/types/src/util/util.ts:387]: OK, limited use so we can probably ignore for now. If it's always a single codepoint we could build a map and ship it -- trivial. Danger with that is that if we use `normalize()` elsewhere then the map could be inconsistent wrt Unicode version.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.262-alpha](https://keyman.com/downloads/releases/alpha/17.0.262)

**Pull request merged**